### PR TITLE
ci: add format and clippy checks

### PR DESCRIPTION
**Pull request by filnet**
taken from Ash
see https://github.com/ash-rs/ash/blob/master/.github/workflows/ci.yml

### CODE REVIEW

**Comment by filnet** [comment]
The clippy and format check should pass when https://github.com/google/shaderc-rs/pull/136 is merged.

PS : I'll rebase this branch once the fix PR is merged.

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -31,3 +31,45 @@ jobs:
         with:
           command: check
 
+  fmt:
+    name: Rustfmt
+    runs-on: ubuntu-latest
```
Comment (filnet) [inline .github/workflows/check.yml:36]: Note the use of `latest` versus `20.02` higher up.

**Comment by filnet** [comment]
I reintroduced some clippy warnings. Trying again...

**Comment by filnet** [comment]
Clippy job now fails because of clippy warnings.
Fixing remaining clippy now...

**Comment by filnet** [comment]
This PR:
- fixes the remaining `uninlined_format_args`
- ci will fail the build if format or clippy fails
- ci now runs on `ubuntu_latest` instead of `ubuntu-22.04`, latest currently is `22.04.1`

Note that ci can now fail spontaneously if, for example, a rust upgrade introduces new default clippy lints.

**Review by antiagainst — APPROVED** [review]

**Pull request merged**